### ISSUE TITLE
Feature Request: RPCS3 Log Analysis

### ISSUE THREAD

**Issue by carlmylo** [hmxmilohax/mhxinfobot#1]
Not sure how possible it is but it would be very nice to have the bot scan any `.log` or `log.gz` that gets sent in `#support` and `#milo-bot-spam`.
Caveats:
* It should only scan after "`Used configuration`" and only if the user has "`Applying custom config`" in their log otherwise it will detect the global options.
* If the user reboots the game multiple times, the configuration will be repeated multiple times. It will need to find the last entry point for RPCS3 loading a custom configuration. It starts at "`Core:`"

The error displays should be formatted like this:
* Use this as the title of the error  
	* Use this for error description.  
	* The method to search for errors. 

## Warnings:
Use ⚠️
* Potentially bad dump.
	* The copy you're using may be bad. Try a better order of Arby's.
	* Search for `Rock Band 3 (USA)`
* Debug Console Mode is off.
	* You currently have Debug Console Mode disabled. You may experience crashes with larger song libraries and longer songs.
	* Search for `Debug Console Mode: false`
* High Memory missing.
	* You're missing the High Memory file. You may experience crashes with larger song libraries and longer songs.
	* Search for `CELL_ENOENT, “/dev_hdd0/game/BLUS30463/USRDIR/dx_high_memory.dta”`
* Weird frame rates settings.
	* You may experience odd frame pacing with your current frame rate settings.
	* If `Frame limit: Auto` or, `Frame limit: 60` isn't found, throw this.
* Emulator in OneDrive folder.
	* I guess just search for any instance of `OneDrive`.
* High Audio Buffer.
	* `Desired Audio Buffer Duration: 100`
* Outdated SPU settings.
	* You are using ASMJIT SPU Decoder. You may be running an older configuration. Please refer to the RPCS3 guide.
	* Search for `SPU Decoder: Recompiler (ASMJIT)`
* Risky Fullscreen settings.
	* You are using a fullscreen mode which may cause issues. Only do this if you know what you're doing.
	* Search for `Exclusive Fullscreen Mode: Enable` and `Exclusive Fullscreen Mode: Automatic`.
* Outdated firmware.
	* You are running older firmware. If you experience a black screen boot, update it.
	* If `SYS: Firmware version: 4.91` isn't found, throw this warning.

## Critical: 
Use 🚫
* No Custom Config found.
	* You are not using a custom configuration, which means certain options will not be available. Please follow the RPCS3 guide.
	* Search for `Selected config: mode=custom config, path=""` exactly.
* MIDI Device failed.
	* Your MIDI device is in use by another application. Please close it so RPCS3 can use it. If you keep getting this error, restart your computer.
	* Search for `log: Could not open port`.
* Bad Driver Delay Wake-Up settings
	* Your Driver Delay Wake-Up setting is set too low. Please increase it or you will crash after a few songs.
	* Search for any value below `Driver Wake-Up Delay: 20` should trigger it. Especially `Driver Wake-Up Delay: 1`
* WCB is off.
	* You have Write Color Buffers disabled. Please follow the RPCS3 guide.
	* Search for `Write Color Buffers: false`
* Missing PS3 Firmware.
	* You seem to be missing firmware. Please follow the RPCS3 guide.
	* Search for `SYS: Missing Firmware`
* Wrong Block Size settings.
	* You are using invalid Block Size settings. Please follow the RPCS3 guide.
	* Search for `SPU Block Size: Giga`
* Wrong Buffer settings
	* You are using unnecessary buffer settings. Please follow the RPCS3 guide.
	* Search for `Write Depth Buffer: true`, `Read Color Buffers: true`, and `Read Depth Buffer: true`
* Bad Shader settings:
	* You are using bad Shader Mode settings. Please follow the RPCS3 guide.
	* Show this if `Shader Mode: Async Shader Recompiler` is not found
* Disconnected network.
	* You have Network Status set to Disconnected. Please follow the RPCS3 guide.
	* Search for `Network Status: Disconnected`
* High Memory crash.
	* You installed the High Memory file but have Debug Console Mode off. This is a guaranteed crash.
	* Search for `Regular file, “/dev_hdd0/game/BLUS30463/USRDIR/dx_high_memory.dta”` while also searching for `Debug Console Mode: false`.
* Bad GPU.
	* You have a graphics device which is below minimum requirements. You will experience severe issues.
	* Search for `Unknown chip with device ID`, `Physical device reports a low amount of allowed deferred descriptor updates`, `Will use graphics queue instead`, and `Your GPU does not support`
* Passthrough Error.
	* RPCS3's USB passthrough system has overloaded. You will need to disconnect one of your PS3 Rock Band instruments.
	* Search for `sys_usbd: Transfer Error:`.
* Unknown Errors (require user info):
	* Detected a fatal error. Please describe what you were doing before crashing.
	* Search for `Thread terminated due to fatal error: Verification failed` and `VM: Access violation reading location`

**Comment by DarkRTA** [comment]
I'd reword passthrough error so its not assuming that's the only cause for a usbd transfer error. More things than just that could cause that.

**Comment by jnackmclain** [comment]
@carlmylo es malo, muy malo
